src/utils/rabbitmq.js

```diff
@@ -1,21 +1,129 @@
 const { Configuration } = require('@schul-cloud/commons');
 const amqp = require('amqplib');
+const logger = require('../logger');
 
 let connection;
 
-const createChannel = async () => {
+const getConnection = async () => {
+	if (!Configuration.get('FEATURE_RABBITMQ_ENABLED') || !Configuration.get('RABBITMQ_URI')) {
+		throw new Error('RabbitMQ is not configured.');
+	}
+
 	if (connection) {
-		const con = await connection; // ensure connection has resolved
-		return con.createChannel();
+		return connection;
 	}
 
-	throw new Error('No RabbitMQ connection available.');
+	try {
+		connection = await amqp.connect(Configuration.get('RABBITMQ_URI'));
+		connection.on('close', () => {
```

Comment:
How is reconnection handled?

Reply:
The getConnection method does a lazy reconnect. When the connection breaks the stored connection is cleared. The next time a connection is required (by calling `getConnection` a new connection is set up.